cpp/DEVELOPER_GUIDE.md

```diff
@@ -275,7 +275,71 @@ void ml_algo(const ML::cumlHandle& handle, ...)
 
 ## Multi GPU
 
-The multi GPU paradigm of cuML is **O**ne **P**rocess per **G**PU (OPG). Each algorithm should be implemented in a way that it can run with a single GPU without any dependencies to any communication library. A multi GPU implementation can assume the following:
-* The user of cuML has initialized MPI and created a communicator that can be used by the ML algorithm.
-* All processes in the MPI communicator call into the ML algorithm cooperatively.
-* The used MPI is CUDA-aware, i.e. it is possible to directly pass device pointers to MPI
+The multi GPU paradigm of cuML is **O**ne **P**rocess per **G**PU (OPG). Each algorithm should be implemented in a way that it can run with a single GPU without any dependencies to any communication library. A multi GPU implementation can use the methods offered by the class `MLCommon::cumlCommunicator` from [cuml_comms_int.hpp](../ml-prims/src/common/cuml_comms_int.hpp) for inter rank/GPU communication. It is the responsibility of the user of cuML to create an initialized instance of `MLCommon::cumlCommunicator`. 
+
+
+E.g. with a CUDA-aware MPI a cuML algorithm could use code like this for 
```

Comment:
I am leaving this in for now and will revisit once our cuda-aware MPI library conda package is available (and thus re-integrate the MPI comms impl).